config/routes.rb

```diff
@@ -13,6 +13,8 @@
 end
 
 Rails.application.routes.draw do
+  admin_constraint = lambda { |request| request.session[:using_admin] }
```

Comment:
There's a little gotcha here, but I don't think its worth doing anything about at this point. Just highlighting for reference.

* Log out
* Go to homepage
* Sign in through sign-in form
* Get redirected back to homepage
* Go directly to `/admin/flipper`
* RouteNotFound error happens

This is because you need to actually go to an`/admin/` page to get `:using_admin` set.

No big deal, but just thought I'd point it out.

Reply:
Had noticed this too. My think was it wasn't a big enough issue to fix especially as it would introduce more complexity in the admin constraint, such as loading users from the DB and checking roles

Reply:
yeah 👍  – :shipit: